bluefog/run/interactive_run.py

```diff
@@ -151,6 +176,45 @@ def _maybe_kill_ipcontroller_process(profile):
         return False
 
 
+def _maybe_kill_ipengine_processes(profile):
+    "Try to kill the ipengine processes through read the pid file."
+    "It only works for local machine case."
+    engine_pids = _get_ipengine_pid_from_file(profile)
+    if engine_pids is None:
+        return
+    for _, pid in engine_pids.items():
+        try:
+            os.kill(pid, signal.SIGINT)
+        except:
+            pass
+
+
+def interrupt_hanged_ipengine_processes(profile):
```

Comment:
Didn't quite understand the logic of this function. Maybe add some comments to help, especially, this is a function exposed to users.

Reply:
Discussed offline. There was a misunderstanding here. "Interrupt" is fine.